### PR TITLE
Add clear-mission RC channel option

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by rmackay9** [comment]
LGTM..

I think we could remove the "if" and this comment, "// failed (I kid you not, read AP_Mission.h)"

**Comment by peterbarker** [comment]
On Thu, 16 Aug 2018, Randy Mackay wrote:

> I think we could remove the "if" and this comment, "// failed (I kid you
> not, read AP_Mission.h)"

Any objection if I invert the return value of that call?

**Comment by peterbarker** [comment]
The comment turned out to be incorrect.  I've added a patch to fix that.

File: libraries/AP_Mission/AP_Mission.h [hunk 1]
```diff
@@ -291,6 +291,8 @@ class AP_Mission {
         _prev_nav_cmd_wp_index(AP_MISSION_CMD_INDEX_NONE),
         _last_change_time_ms(0)
     {
+        _singleton = this;
```
Comment (WickedShell) [inline libraries/AP_Mission/AP_Mission.h]: Typical pattern is to add a HAL_SITL guarded check to this to ensure that `_singleton` was null when we created the object.

**Comment by peterbarker** [comment]
On Sun, 2 Sep 2018, WickedShell wrote:

> Typical pattern is to add a HAL_SITL guarded check to this to ensure that

Prove it :-)

We've got *lots* of patterns to choose from :-)

**Comment by peterbarker** [comment]
@WickedShell Added SITL sanity check for mission singletonness.

**Pull request merged**